telemetry/specs/INT.mdk

```diff
@@ -384,45 +384,95 @@ Considering all these factors, the INT specification requires that an INT
-transported over IPv4, it is possible to assign a zero Checksum, causing the 
-receiver to ignore the value of the Checksum field. For UDP over IPv6, there
-are specific use cases in which it is possible to assign a zero Checksum 
-(as defined in RFC 6936).
-
-If an L4 Checksum update is required, an INT source/transit switch may perform
-it in one of two ways:
-
-* Update the L4 Checksum field such that the new value is equal to the Checksum 
+encapsulation header that includes an L4 header, such as VXLAN.
+
+In theory, since each INT transit hop modifies the Layer 4 payload, it should
+update the Layer 4 checksum. Layer 4 checksum update at transit hops is
+necessary to protect the integrity of INT metadata between the INT source
+and INT sink. If integrity of INT metadata is to be protected by Layer 4
+checksum, each INT transit hop must perform incremental checksum calculation
+as complete checksum calculation at a transit hop will mask any corruption
+in the INT metadata inserted by previous INT hops.
+
+Layer 4 checksum updates at INT hops may or may not be required to protect
+the integrity of the Layer 4 payload.
+
+* In the case of encapsulation protocols, the integrity of the encapsulated
+payload is covered by the inner Layer 4 checksum. Hence, UDP-based
+encapsulation protocols such as VXLAN and Geneve allow for outer UDP
+checksum to be set to zero in accordance with RFC 768 and RFC 6936,
+bypassing Layer 4 checksum validation at the receiving tunnel endpoint.
+While TCP does not define a specific checksum value to indicate that
+checksum is not to be validated at the receiver, it is possible for
+certain implementations of TCP-based encapsulation protocols
+such as Stateles Transport Tunneling Protocol (STT) to also ignore outer
+Layer 4 checksum at the receiving tunnel endpoint based on a similar rationale
+that the integrity of the encapsulated payload is covered by the inner Layer 4
+checksum.
+* If INT is enabled on probe packets, Layer 4 checksum update is required
+if integrity of INT metadata and other information carried in probe packets
+is desired.
+* If INT is enabled on real data packets and INT source is not the original
+traffic source, all additions to the packet at the INT source and transit hops
+are removed at the INT sink, making the original Layer 4 checksum valid after
+the INT sink.
+* If INT is enabled on real data packets and INT source is also the Layer 4
+traffic source, it may compute the Layer 4 checksum on the entire Layer 4
+payload including INT headers. In this case, if INT sink is also the Layer 4
+destination, every INT transit hop must update the Layer 4 checksum.
+If the INT sink is not the Layer 4 destination, transit hops may not update
+the checksum (unless INT metadata integrity is to be protected), the
+INT sink must update the Layer 4 checksum upon removing INT headers and
+metadata, before forwarding the data packet to the destination.
```

Comment:
This paragraph is difficult to decipher. I have tried multiple times and cannot tell what it is saying, or what requirements it is imposing.

Why does it matter whether the INT sink is the L4 destination or not?

Language such as "transit hops may not update the checksum" is confusing. It should be clear from the received packet whether checksum integrity must be preserved, or the checksum must be left unaltered. It is not up to the transit hop to decide, is it?